panoply/datasource.py

```diff
@@ -20,6 +24,27 @@ def __init__(self, source, options={}):
         self.source = source
         self.options = options
 
+    @abstractmethod
+    def read(self, batch_size=None) -> List[RecordGroup]:
+        """
+        Reads data from the sources and returns it as record group
+        """
+
+    @classmethod
+    @abstractmethod
+    def get_resource(cls, resource_id: str, source, options) -> Resource:
```

Comment:
```suggestion
    def get_resource(cls, resource_id: str, source, options={}) -> Resource:
```

Add default value for options